rpc/rpcserver.go

```diff
@@ -83,35 +105,54 @@ func (r *rpcServer) UpdateGateway(ctx context.Context, in *rpcapi.UpdateGatewayR
 		broadcastAcks = 1
 	}
 
+	gtwManager, err := gateway.NewManager(ctx, in.GatewayAddresses, connAcks)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { gtwManager.Close() }()
 	b, err := broadcaster.New(
-		in.GatewayAddresses,
+		gtwManager.Connections(),
 		in.DisableBroadcast,
-		broadcaster.DefaultConnTimeout,
-		uint(connAcks),
 		broadcaster.DefaultBroadcastTimeout,
 		uint(broadcastAcks),
 	)
 	if err != nil {
 		return nil, err
 	}
 
+	verifier, err := service.CreateChallengeVerifier(gtwManager.Connections())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create ATX provider: %w", err)
+	}
+
+	// Close the old connections and save the new manager.
+	// The temporary manager is nil-ed to avoid closing connections in defer.
+	r.gtwManager.Close()
+	r.gtwManager, gtwManager = gtwManager, nil
```

Comment:
I realized it can be improved so that the nil checks can be removed:
```go
// Swap the new and old gateway managers. 
// The old one will be closed in defer. 
r.gtwManager, gtwManager = gtwManager, r.gtwManager
```